### ISSUE TITLE
resource property of type number without default

### ISSUE THREAD

**Issue by OlafConijn** [eduardomourar/cloudformation-cli-typescript-plugin#21]
hi, seems like having a number without a default value results in an internal error when not specified. i would expect the value to be undefined.

**Comment by eduardomourar** [comment]
I will have a look. I believe the framework is casting to number (or bigint for that matter) but only if different than undefined or null. https://github.com/eduardomourar/cloudformation-cli-typescript-plugin/blob/26497282b6189f599fa6f98b99bf19592be19d82/src/recast.ts#L40-L42

**Comment by eduardomourar** [comment]
I see the problem now. We are receiving an empty string from CloudFormation and that is being converted to 0.
```json
{
  "message": "Error: 1 validation error detected: Value '0' at 'passwordReusePrevention' failed to satisfy constraint: Member must have value greater than or equal to 1",
  "status": "FAILED",
  "errorCode": "InternalFailure"
}
```

**Comment by eduardomourar** [comment]
@OlafConijn , you can check the unit test failing here: https://github.com/eduardomourar/cloudformation-cli-typescript-plugin/pull/22/checks?check_run_id=999912293. is there any other problem besides that?

**Comment by OlafConijn** [comment]
no, i think thats it :)

**Comment by eduardomourar** [comment]
Fixed on release v0.3.1
